functions.php

```diff
@@ -7,30 +7,24 @@ function getDB () {
     $plant_query = $db->prepare("SELECT * FROM `plants` LEFT JOIN `plant_types` ON `plants`.`type` = `plant_types`.`id`;");
     $plant_query->execute();
 
-    return $plant_query;
-}
-
-function makeAllTiles ($query) {
-    $plantEntry = getDataEntry($query);
-    while ($plantEntry) {
-        makePlantEntryTile($plantEntry);
-        $plantEntry = getDataEntry($query);
-    }
-}
-
-function getDataEntry(object $dataInput): array {
-    if (!empty($dataInput)) {
-        if (gettype($dataInput) === 'object') {
-            $entry = $dataInput->fetch();
-            return $entry;
+    $plant_data = $plant_query->fetchAll();
+    if (!empty($plant_data)) {
+        if (gettype($plant_data) === 'array') {
+            return $plant_data;
         } else {
-            echo 'Please input the correct data type';
-        }
+             echo 'Unexpected error. Please refresh page.';
+         }
     } else {
         echo 'There is no data for this collection.';
     }
 }
 
+function makeAllTiles ($data) {
```

Comment:
You have makeAllTiles simply refer to your other function makePlantEntryTile but with a foreach loop yet only makeAllTiles is being used in your index file. Any reason to not simply combine them into one?